analyzer-java/src/test/java/com/infosupport/ldoc/analyzerj/AnalysisVisitorTest.java

```diff
@@ -554,6 +554,29 @@ void annotation_declaration_with_field_no_default() {
 
     assertEquals(expected, typeDescription);
   }
+
+
+  @Test
+  void variable_declaration_with_annotation() {
+    String code = """
+        class Test {
+        void main(String[] args) {
+            @SuppressWarnings
+            int x = 0;
+          }
+          }
+        """;
+
+    List<Description> expected = List.of(new TypeDescription.Builder(TypeType.CLASS, "Test")
+            .withMembers(new MethodDescription(
+                new MemberDescription("main", 0, List.of(), null),
+                null, List.of(new ParameterDescription("java.lang.String[]", "args", List.of())),
+                List.of()
+            ))
+        .build());
+
+    assertIterableEquals(parse(code), expected);
```

Comment:
Suggest `parseFragment` here instead of `parse` to make the test a bit shorter. You will need to put at least one statement that's not ignored (say, `return x;`).